From: Felipe Sateler <fsateler@gmail.com>
Date: Thu, 15 Apr 2010 14:47:43 -0400
Subject: [PATCH] Change the textdomaindir to match debian installation

===================================================================
---
 checkinstall |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/checkinstall b/checkinstall
index 41a7978..d2ba87e 100755
--- a/checkinstall
+++ b/checkinstall
@@ -65,7 +65,7 @@ INSTALLDIR=`followlinks "$0"`
 INSTALLDIR=`dirname "$INSTALLDIR"`/..
 
 # gettext variables
-export TEXTDOMAINDIR="$INSTALLDIR"/lib/checkinstall/locale
+export TEXTDOMAINDIR=/usr/share/locale
 export TEXTDOMAIN=checkinstall
 
 # .spec file path
-- 
