Description: Don't mark directories in etc as conf files
Forwarded: http://bugtrack.izto.org/show_bug.cgi?id=35
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=655453
Author: Zsolt Müller <muzso@muzso.hu>
Last-Update: 2012-04-01
--- a/checkinstall
+++ b/checkinstall
@@ -2609,8 +2609,8 @@
 fi
 
 # Tag files in /etc to be conffiles
-find $BUILD_DIR/etc 2> /dev/null | sed -e "s,$BUILD_DIR,," | \
-	grep -v '^/etc$' > $BUILD_DIR/DEBIAN/conffiles
+find -L $BUILD_DIR/etc -type f 2> /dev/null | sed -e "s,$BUILD_DIR,," | \
+        > $BUILD_DIR/DEBIAN/conffiles
 
 # The package will be saved here (ignores <epoch>: prefix):
 DEBPKG="${DIRECTORIO_FUENTE}/${NAME}_`echo ${VERSION} | sed -e 's/^[0123456789]*://'`-${RELEASE}_${ARCHITECTURE}.deb"
