Index: checkinstall/checkinstall
===================================================================
--- checkinstall.orig/checkinstall	2010-04-15 14:04:53.000000000 -0400
+++ checkinstall/checkinstall	2010-04-15 14:07:54.000000000 -0400
@@ -65,7 +65,7 @@
 INSTALLDIR=`dirname "$INSTALLDIR"`/..
 
 # gettext variables
-export TEXTDOMAINDIR="$INSTALLDIR"/lib/checkinstall/locale
+export TEXTDOMAINDIR=/usr/share/locale
 export TEXTDOMAIN=checkinstall
 
 # .spec file path
