Index: checkinstall/checkinstall
===================================================================
--- checkinstall.orig/checkinstall	2008-06-25 18:20:06.000000000 -0400
+++ checkinstall/checkinstall	2008-06-25 18:22:30.000000000 -0400
@@ -45,7 +45,7 @@
 
 
 # gettext variables
-export TEXTDOMAINDIR=/usr/local/lib/checkinstall/locale
+export TEXTDOMAINDIR=/usr/share/locale
 export TEXTDOMAIN=checkinstall
 
 # .spec file path
