Index: checkinstall/checkinstall
===================================================================
--- checkinstall.orig/checkinstall	2008-06-25 18:22:30.000000000 -0400
+++ checkinstall/checkinstall	2008-06-25 18:22:44.000000000 -0400
@@ -386,13 +386,13 @@
 ckversion
 echo
 
-if ! [ -f /usr/local/lib/checkinstall/checkinstallrc ]; then
-   echog "The checkinstallrc file was not found at:\n/usr/local/lib/checkinstall/checkinstallrc"
+if ! [ -f /etc/checkinstallrc ]; then
+   echog "The checkinstallrc file was not found at:\n/etc/checkinstallrc"
    echo
    echog "Assuming default values."
 else
 # Get our default settings from the rc file
-source /usr/local/lib/checkinstall/checkinstallrc
+source /etc/checkinstallrc
 fi
 
 
