Index: checkinstall/checkinstall
===================================================================
--- checkinstall.orig/checkinstall	2010-04-15 14:07:54.000000000 -0400
+++ checkinstall/checkinstall	2010-04-15 14:08:54.000000000 -0400
@@ -492,7 +492,7 @@
 ckversion
 echo
 
-CHECKINSTALLRC=${CHECKINSTALLRC:-${INSTALLDIR}/checkinstallrc}
+CHECKINSTALLRC=${CHECKINSTALLRC:-/etc/checkinstallrc}
 if ! [ -f $CHECKINSTALLRC ]; then
    echog "The checkinstallrc file was not found at:\n$CHECKINSTALLRC"
    echo
