Index: checkinstall/checkinstall
===================================================================
--- checkinstall.orig/checkinstall	2008-06-25 19:19:55.000000000 -0400
+++ checkinstall/checkinstall	2008-06-25 19:20:47.000000000 -0400
@@ -375,6 +375,17 @@
 
 }
 
+# This function is used to properly escape shell commands, so
+# we don't interpret the install command twice. This is useful
+# for making commands like 'make CC="gcc -some -options" install'
+# work as expected.
+
+function shell_escape() {
+	for str in "$@" ; do
+		echo -n "\"$str\" "
+	done;
+	echo
+}
 
 
                       #################################
@@ -744,7 +755,7 @@
 # See if we have and install command
 shift
 
-[ "$1" = "" ] && set -- "make install"
+[ "$1" = "" ] && set -- make install
 
 # End of argument parsing
 
@@ -884,8 +895,8 @@
 
 
 
-INSTALLCMD="$@"
-: ${INSTALLCMD:='make install'}
+INSTALLCMD=("$@")
+[ -z "${INSTALLCMD[*]}" ] && INSTALLCMD=(make install)
 
 # The editor we use to review the file list and .spec or control file
 if ! [ -x "$VISUAL" ]; then
@@ -1378,8 +1389,11 @@
 #!/bin/sh
 
 cd "$DIRECTORIO_FUENTE"
-$INSTALLCMD
+EOF
+
+shell_escape "${INSTALLCMD[@]}" >> $TMP_SCRIPT
 
+cat << EOF >> $TMP_SCRIPT
 # Report success or failure
 if [ \$? -eq 0 ]; then
    exit 0
@@ -1394,7 +1408,7 @@
 
 
 echo
-echogn "Installing with %s..." "$INSTALLCMD"
+echogn "Installing with %s..." "${INSTALLCMD[@]}"
 
 
 #
