Index: checkinstall/checkinstall
===================================================================
--- checkinstall.orig/checkinstall	2008-06-25 19:20:47.000000000 -0400
+++ checkinstall/checkinstall	2008-06-25 19:47:09.000000000 -0400
@@ -1813,6 +1813,41 @@
                        # End of Identify shared libraries # 
 ##############################################################################
 
+##############################################################################
+                          # Identify kernel modules #
+			  ###########################
+
+# Find out if we installed a kernel module, and if so run depmod in the postinst
+# and delete the files depmod touches (they conflict with other packages)
+# These files are modules.* inside the kernel modules directory.
+
+DEPFILES=$(cat ${TMP_DIR}/newfiles | grep -E 'modules\.(dep|pcimap|usbmap|ccwmap|ieee1394map|isapnpmap|inputmap|ofmap|seriomap|alias|symbols)' )
+if [ ! -z "$DEPFILES" ] ; then
+	echo
+	echog "Kernel modules found. Calling depmod in the postinstall script"
+	# Delete references to the depmod generated files
+	sed -r -i -e '\,modules\.(dep|pcimap|usbmap|ccwmap|ieee1394map|isapnpmap|inputmap|ofmap|seriomap|alias|symbols),d' ${TMP_DIR}/newfiles
+	# And the files themselves
+	for file in $DEPFILES ; do
+		rm ${BUILD_DIR}/$file
+	done
+	# Find out the kernel version.
+	# Ugly hack, but works in Debian and should on many other systems.
+	# The file must be /any/dir/<version>/modules.*
+	KVER=$(echo $DEPFILES | awk '{print $1}' | sed -e 's|/modules\..*||' | xargs basename )
+	# Add depmod to the postinst
+	# This will be redundant if we already have a postinstall script.
+	# Nevertheless, it is harmless.
+	echo "#!/bin/sh" >> ${DIRECTORIO_FUENTE}/postinstall-pak
+	echo "#" >> ${DIRECTORIO_FUENTE}/postinstall-pak
+	echo "# postinstall script, created by checkinstall-${CHECKINSTALL_VERSION}" >> ${DIRECTORIO_FUENTE}/postinstall-pak
+	echo "#" >> ${DIRECTORIO_FUENTE}/postinstall-pak
+	echo "depmod $KVER" >> ${DIRECTORIO_FUENTE}/postinstall-pak
+fi
+
+                        ##################################
+		        # End of Identify kernel modules #
+##############################################################################
 
 cd $DIRECTORIO_FUENTE
 
