Index: checkinstall/checkinstall
===================================================================
--- checkinstall.orig/checkinstall	2008-06-25 19:47:25.000000000 -0400
+++ checkinstall/checkinstall	2008-06-25 19:47:26.000000000 -0400
@@ -1542,6 +1542,22 @@
 	   cp /${TMP_DIR}/newfiles.tmp /${TMP_DIR}/newfiles
 	done
 
+	# Find files created under /home
+	grep '^/home' ${TMP_DIR}/newfile > /${TMP_DIR}/unwanted
+	if [ $? -eq 0 ]; then
+	   echo
+	   echog "Some of the files created by the installation are inside the home directory: /home"
+	   echo
+	   echog "You probably don't want them to be included in the package."
+   	   echogn "Do you want me to list them? "
+	   if yorn "n"; then
+	       $PAGER ${TMP_DIR}/unwanted
+           fi
+	   echogn "Should I exclude them from the package? (Saying yes is a good idea) "
+	   if yorn "n"; then
+	       sed -i -e '/^\/home/d' ${TMP_DIR}/newfiles
+           fi
+	fi
 
 	## Find any files created in `pwd`. We probably don't want them here
 
