Index: checkinstall/checkinstall
===================================================================
--- checkinstall.orig/checkinstall	2008-06-25 20:31:26.000000000 -0400
+++ checkinstall/checkinstall	2008-06-25 22:03:51.000000000 -0400
@@ -130,7 +130,7 @@
    echog "--pkgaltsource=<altsource>     Set alternate source location"
    echog "--pakdir=<directory>           The new package will be saved here"
    echog "--maintainer=<email addr>      The package maintainer (.deb)"
-   echog "--provides=<list>              Features provided by this package (.rpm)"
+   echog "--provides=<list>              Features provided by this package"
    echog "--requires=<list>              Features required by this package"
    echog "--rpmflags=<flags>             Pass this flags to the rpm installer"
    echog "--rpmi                         Use the -i flag for rpm when installing a .rpm"
@@ -1283,8 +1283,10 @@
       echog "7 -  Architecture: [ %s ]" "$ARCHITECTURE"
       echog "8 -  Source location: [ %s ]" "$SOURCE"
       echog "9 -  Alternate source location: [ %s ]" "$ALTSOURCE"
-      echog "10 - Requires: [ %s ]" "$REQUIRES"
-      [ "$CK_REDHAT" ] && echog "11 - Provides: [ %s ]" "$PROVIDES"
+      [ "$CK_REDHAT" -o "$CK_DEBIAN" ] &&
+          echog "10 - Requires: [ %s ]" "$REQUIRES"
+      [ "$CK_REDHAT" -o "$CK_DEBIAN" ] &&
+          echog "11 - Provides: [ %s ]" "$PROVIDES"
 
 
       echo
@@ -1350,7 +1352,7 @@
 	    }
             ;;
         10)
-            [ "$CK_REDHAT" ] && {
+            [ "$CK_REDHAT" -o "$CK_DEBIAN" ] && {
             echog "Enter the additional requirements: "
             echo -n ">> "
             read REQUIRES
@@ -1358,9 +1360,11 @@
             ;;
         11)
             # 01-12-06 UKo: new feature
+	    [ "$CK_REDHAT" -o "$CK_DEBIAN" ] && {
             echog "Enter the provided features: "
             echo -n ">> "
             read PROVIDES
+            }
             ;;
       esac
    done
@@ -2405,6 +2409,7 @@
 Architecture: $ARCHITECTURE
 Version: ${VERSION}-${RELEASE}
 Depends: $REQUIRES
+Provides: $PROVIDES
 Description: $SUMMARY
 EOF
 
