Index: checkinstall/Makefile
===================================================================
--- checkinstall.orig/Makefile	2008-06-25 20:03:46.000000000 -0400
+++ checkinstall/Makefile	2008-06-25 20:04:17.000000000 -0400
@@ -19,11 +19,11 @@
 			;; \
 		esac ; \
 	done	
-	make -C installwatch-0.7.0beta5
+	$(MAKE) -C installwatch-0.7.0beta5
 	
 install: all
 	export
-	make -C installwatch-0.7.0beta5 install
+	$(MAKE) -C installwatch-0.7.0beta5 install
 	
 	mkdir -p $(BINDIR)
 	install checkinstall makepak $(BINDIR)
@@ -58,4 +58,4 @@
 	for file in locale/checkinstall-*.mo ; do \
 		rm -f $${file} ; \
 	done
-	make -C installwatch-0.7.0beta5 clean
+	$(MAKE) -C installwatch-0.7.0beta5 clean
