Index: checkinstall/installwatch-0.7.0beta5/Makefile
===================================================================
--- checkinstall.orig/installwatch-0.7.0beta5/Makefile	2008-06-25 20:05:55.000000000 -0400
+++ checkinstall/installwatch-0.7.0beta5/Makefile	2008-06-25 20:06:00.000000000 -0400
@@ -30,7 +30,7 @@
 	if [ -r $(LIBDIR)/installwatch.so ]; then \
 		rm -f  $(LIBDIR)/installwatch.so; \
 	fi
-	install -s installwatch.so $(LIBDIR)
+	install installwatch.so $(LIBDIR)
 	
 	sed -e "s|#PREFIX#|$(PREFIX)|" < installwatch > $(BINDIR)/installwatch
 	chmod 755 $(BINDIR)/installwatch
