Index: checkinstall/checkinstall
===================================================================
--- checkinstall.orig/checkinstall	2009-08-20 23:33:01.000000000 -0400
+++ checkinstall/checkinstall	2009-08-20 23:34:26.000000000 -0400
@@ -1141,7 +1141,7 @@
    ARCHITECTURE=`uname -m`
    echo $ARCHITECTURE | grep -e "i[3456]86" &> /dev/null
    [ $? -eq 0 ] && ARCHITECTURE=i386  # Arch will be "i386" for any of
-fi                                    # i386, i486, i586 or i686.
+                                      # i386, i486, i586 or i686.
                                       # You can change this with "--arch"
 
    # Fix the PowerPC & Amd64 architecture description if we're on Debian
@@ -1149,6 +1149,7 @@
    if [ "$CK_DEBIAN" = "1" ] ; then
       ARCHITECTURE=`dpkg-architecture -qDEB_HOST_ARCH`
    fi
+fi
 
 if [ -f "$SPEC_PATH" ]; then 
    SUMMARY=`getparameter "^[Ss]ummary"`
