Index: checkinstall/checkinstallrc-dist
===================================================================
--- checkinstall.orig/checkinstallrc-dist	2008-06-25 22:28:59.000000000 -0400
+++ checkinstall/checkinstallrc-dist	2008-06-25 23:35:38.000000000 -0400
@@ -16,7 +16,7 @@
 DEBUG=0
 
 # Location of the "installwatch" program
-INSTALLWATCH_PREFIX="/usr/local"
+INSTALLWATCH_PREFIX="/usr"
 INSTALLWATCH=${INSTALLWATCH_PREFIX}/bin/installwatch
 
 # Location of the makepkg program. "makepak" is the default, and is 
@@ -39,7 +39,7 @@
 BASE_TMP_DIR=/var/tmp   ##  Don't set this to /tmp or / !! 
 
 # Where to place the installed document files
-DOC_DIR=""
+DOC_DIR="/usr/share/doc"
 
 # Default architecture type (Leave it empty to allow auto-guessing)
 ARCHITECTURE=""
@@ -49,7 +49,7 @@
 #   R : RPM
 #   D : Debian
 
-INSTYPE=""
+INSTYPE="D"
 
 # Storage directory for newly created packages
 # By default they will be stored at the default
@@ -72,16 +72,16 @@
 
 # Show Slackware package installation script while it runs? Again, useful if
 # it's an interactive script
-SHOW_SLACK_INSTALL=0
+SHOW_SLACK_INSTALL=1
 
 # Automatic deletion of "doc-pak" upon termination?
-DEL_DOCPAK=1
+DEL_DOCPAK=0
 
 # Automatic deletion of the spec file?
-DEL_SPEC=1
+DEL_SPEC=0
 
 # Automatic deletion of "description-pak"?
-DEL_DESC=1
+DEL_DESC=0
 
 # Automatically strip all ELF binaries?
 STRIP_ELF=1
@@ -108,7 +108,9 @@
 AUTODOINST=1
 
 # Are we going to use filesystem translation?
-TRANSLATE=1            
+# Note: checkinstall, as of 1.6.1 doesn't handle the at family of syscalls.
+# This means that translation doesn't work as coreutils uses them.
+TRANSLATE=0
 
 # Reset the owner/group of all files to root.root?
 RESET_UIDS=0               
@@ -136,4 +138,4 @@
 REVIEW_CONTROL=0
       
 # Install the package or just create it?
-INSTALL=0
+INSTALL=1
